File: 06-Sum-Square-Difference/sumsq.c
#include <stdio.h>

// Find the difference between (1+2+3+...+100)^2 and (1^2+2^2+3^2+...+100^2)
// I used another math trick here:
// The sum up to a number can be found using Gauss' Trick:
// 1 + 2 + 3 + ... + 10 -> (1+9) + (2+8) + (3+7) + (4+6) + 10 + 5 -> 5*10 +5
// In general : (n/2)(n+1)

int main() {
    long sqsum = 50 * 101;
    sqsum *=sqsum;
    printf("The square of the sum is: %lu\n", sqsum);
    long sumsq = 0;
    for(long i = 1; i <= 100; i++) {
        sumsq += i * i;
    }
    printf("The sum of the squares is: %lu\n", sumsq);
    printf("Difference: %lu\n", sqsum - sumsq);
    return 0;
}

File: 06-Sum-Square-Difference/sumsq.py
# The sum of the squares of the first ten natural numbers is,
# 12 + 22 + ... + 102 = 385
# The square of the sum of the first ten natural numbers is,
# (1 + 2 + ... + 10)2 = 552 = 3025
# Hence the difference between the sum of the squares
# of the first ten natural numbers and the square of the sum is
# Find the difference between the sum of the squares of the first
# one hundred natural numbers and the square of the sum.

import PIL
import math

nums = range(1, 101)
numsq = []

for i in nums:
    numsq.append(i*i)

sum1 = (sum(nums)**2)
sum2 = (sum(numsq))

print("The sum squared is: " + str(sum1))
print("The sum of the squares is: " + str(sum2))

print("The difference is: " + str(abs(sum2-sum1)))
